src/uu/tr/src/operation.rs

```diff
@@ -0,0 +1,569 @@
+    unicode_table::SPACE,
+];
+static BLANK: &'static [char] = &[unicode_table::SPACE, unicode_table::HT];
+
+pub enum Sequence {
+    Char(char),
+    CharRange(Box<dyn Iterator<Item = char>>),
+    CharStar(char),
+}
+
+impl Sequence {
+    pub fn parse_set_string(input: &str) -> Vec<Sequence> {
+        many0(alt((
+            alt((
+                Sequence::parse_char_range_octal_leftright,
+                Sequence::parse_char_range_octal_left,
+                Sequence::parse_char_range_octal_right,
+                Sequence::parse_char_range_backslash_collapse,
+                Sequence::parse_char_range,
+                Sequence::parse_char_star,
+                Sequence::parse_char_repeat,
+            )),
+            alt((
+                Sequence::parse_alnum,
+                Sequence::parse_alpha,
+                Sequence::parse_blank,
+                Sequence::parse_control,
+                Sequence::parse_digit,
+                Sequence::parse_graph,
+                Sequence::parse_lower,
+                Sequence::parse_print,
+                Sequence::parse_punct,
+                Sequence::parse_space,
+                Sequence::parse_upper,
+                Sequence::parse_xdigit,
+                Sequence::parse_char_equal,
+                // NOTE: This must be the last one
+            )),
+            alt((
+                Sequence::parse_octal,
+                Sequence::parse_backslash,
+                Sequence::parse_char,
+            )),
+        )))(input)
+        .map(|(_, r)| r)
+        .unwrap()
+    }
+
+    pub fn dissolve(self) -> Box<dyn Iterator<Item = char>> {
+        match self {
+            Sequence::Char(c) => Box::new(std::iter::once(c)),
+            Sequence::CharRange(r) => r,
+            Sequence::CharStar(c) => Box::new(std::iter::repeat(c)),
+        }
+    }
+
+    /// Sequence parsers
+
+    fn parse_char(input: &str) -> IResult<&str, Sequence> {
+        anychar(input).map(|(l, r)| (l, Sequence::Char(r)))
+    }
+
+    fn parse_backslash(input: &str) -> IResult<&str, Sequence> {
+        preceded(tag("\\"), anychar)(input).map(|(l, a)| {
+            let c = match a {
+                'a' => Sequence::Char(unicode_table::BEL),
+                'b' => Sequence::Char(unicode_table::BS),
+                'f' => Sequence::Char(unicode_table::FF),
+                'n' => Sequence::Char(unicode_table::LF),
+                'r' => Sequence::Char(unicode_table::CR),
+                't' => Sequence::Char(unicode_table::HT),
+                'v' => Sequence::Char(unicode_table::VT),
+                x => Sequence::Char(x),
+            };
+            (l, c)
+        })
+    }
+
+    fn parse_octal(input: &str) -> IResult<&str, Sequence> {
+        map_opt(
+            preceded(tag("\\"), recognize(many_m_n(1, 3, one_of("01234567")))),
+            |out: &str| {
+                u32::from_str_radix(out, 8)
+                    .map(|u| Sequence::Char(char::from_u32(u).unwrap()))
```

Comment:
```suggestion
                    .map(|u| Sequence::Char(std::char::from_u32(u).unwrap()))
```
`char::from_u32` was recently added, `std::char::from_u32` works for earlier versions too. There are other places where you have to make this change, too.